File: my-app/src/types/submission.ts
export interface SubmissionCreateResponse {
  id: string;
  task_id: string;
  student_id: string;
  version: number;
  content_type: string;
  status: string;
  submitted_at: string;
  task_title: string;
}

export interface DimensionScore {
  name: string;
  score: number;
  max_score: number;
  comment: string;
}

export interface GradingFeedback {
  dimensions: DimensionScore[];
  improvements: string[];
  highlights: string[];
  overall_comment: string;
}

export interface SubmissionDetail {
  id: string;
  task_id: string;
  task_title: string;
  version: number;
  content_type: string;
  status: string;
  score: number | null;
  feedback: GradingFeedback | null;
  submitted_at: string;
  graded_at: string | null;
}

export interface SubmissionListResponse {
  items: SubmissionDetail[];
  student_name?: string;
}

export interface SubmissionContentResponse {
  submission_id: string;
  filename: string;
  content: string;
  content_type: string;
  file_extension: string;
}

File: backend/schemas/submission.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class SubmissionCreateResponse(BaseModel):
    id: uuid.UUID
    task_id: uuid.UUID
    student_id: uuid.UUID
    version: int
    content_type: str
    status: str
    submitted_at: datetime
    task_title: str


class SubmissionDetail(BaseModel):
    id: uuid.UUID
    task_id: uuid.UUID
    task_title: str
    version: int
    content_type: str
    status: str
    score: float | None
    suggestion: str | None
    submitted_at: datetime
    graded_at: datetime | None


class SubmissionListResponse(BaseModel):
    items: list[SubmissionDetail]
    student_name: str | None = None


class SubmissionContentResponse(BaseModel):
    submission_id: uuid.UUID
    filename: str
    content: str
    content_type: str
    file_extension: str
